modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -41,6 +42,74 @@ def bgs_service
           BGS::Services.new(external_uid: target_veteran.participant_id,
                             external_key: target_veteran.participant_id)
         end
+
+        def map_claims(bgs_claims:, lighthouse_claims:)
+          mapped_claims = bgs_claims[:benefit_claims_dto][:benefit_claim].map do |bgs_claim|
+            matching_claim = find_bgs_claim_in_lighthouse_collection(
+              lighthouse_collection: lighthouse_claims,
+              bgs_claim: bgs_claim
+            )
+            if matching_claim
+              lighthouse_claims.delete(matching_claim)
+              BGSToLighthouseClaimsMapperService.process(bgs_claim: bgs_claim, lighthouse_claim: matching_claim)
+            else
+              BGSToLighthouseClaimsMapperService.process(bgs_claim: bgs_claim)
+            end
+          end
+
+          lighthouse_claims.each do |remaining_claim|
+            # if claim wasn't matched earlier, then this claim is in a weird state where
+            #  it's 'established' in Lighthouse, but unknown to BGS.
+            #  shouldn't really ever happen, but if it does, skip it.
+            next if remaining_claim.status.casecmp?('established')
+
+            mapped_claims << BGSToLighthouseClaimsMapperService.process(lighthouse_claim: remaining_claim)
+          end
+
+          mapped_claims
+        end
+
+        def find_bgs_claim_in_lighthouse_collection(lighthouse_collection:, bgs_claim:)
+          lighthouse_collection.find { |lighthouse_claim| lighthouse_claim.evss_id == bgs_claim[:benefit_claim_id] }
+        end
+
+        def find_lighthouse_claim!(claim_id:)
+          lighthouse_claim = ClaimsApi::AutoEstablishedClaim.get_by_id_or_evss_id(claim_id)
+
+          if looking_for_lighthouse_claim?(claim_id: claim_id) && lighthouse_claim.blank?
```

Comment:
is there a reason we need to check whether the `claim_id` is the lighthouse id vs EVSS id?

Reply:
nvm, I forgot that `evss_id` is misleading and serves as the main identifier in the BGS call, as well. Maybe just add a comment explaining that bit of business logic, since it's not obvious from the variable/method names here